src/knot_thing_main.c

```diff
@@ -153,13 +153,103 @@ int8_t knot_thing_config_data_item(uint8_t sensor_id, uint8_t event_flags,
 
 int8_t knot_thing_run(void)
 {
+	uint8_t i = 0, uint8_val = 0, comparison = 0, uint8_buffer[KNOT_DATA_RAW_SIZE];
+	int32_t int32_val = 0, multiplier = 0;
+	uint32_t uint32_val = 0;
 	/* TODO: Monitor messages from network */
 
 	/*
-	 * TODO: For all registered data items: verify if value
+	 * For all registered data items: verify if value
 	 * changed according to the events registered.
 	 */
 
+	// TODO: add timer events
+	for (i = 0; i < KNOT_THING_DATA_MAX; i++)
+	{
+		if (data_items[i].value_type == KNOT_VALUE_TYPE_RAW) 
+		{
+			// Raw supports only KNOT_EVT_FLAG_CHANGE
+			if ((KNOT_EVT_FLAG_CHANGE & data_items[i].event_flags) == 0)
+				continue;
+
+			if (data_items[i].functions.raw_f.read == NULL)
+				continue;
+
+			if (data_items[i].functions.raw_f.read(&uint8_val, uint8_buffer) < 0)
+				continue;
+
+			if (memcmp(data_items[i].last_value_raw, uint8_buffer, uint8_val) == 0)
```

Comment:
Is there any freak dependency for AVR or other embedded platform when using memcmp?